src/xrpld/app/tx/detail/Payment.cpp

```diff
@@ -420,6 +444,75 @@ Payment::doApply()
             terResult = tecPATH_DRY;
         return terResult;
     }
+    else if (bMPTDirect)
+    {
+        JLOG(j_.trace()) << " saDstAmount=" << saDstAmount.getFullText();
+
+        if (auto const ter =
+                requireAuth(view(), saDstAmount.get<MPTIssue>(), account_);
+            ter != tesSUCCESS)
+            return ter;
```

Comment:
I don't think it makes sense to check if the sender is authorized to hold the MPT.
1. If they have never been authorized they'll have a zero balance. The "sufficient balance" check will fail, and no funds will be sent.
2. If they have previously been authorized, got a balance, and were later unauthorized, they will still have a balance, so why wouldn't they be allowed to get rid of it by sending it to someone? (Note that `MPTAuthorize` does not check for a balance.)
3. If they're the issuer, they won't have an `MPToken` object, but they can obviously issue / mint more tokens any time they want.

```suggestion
```

Reply:
It's possible that the holder has been unauthorized, which means they could still have remaining balance. In this case, the issuer needs to clawback from the holder. So it's still possible that the sender is not unauthorized but still own funds.

Reply:
Ok, so my assumption here is wrong:

> so why wouldn't they be allowed to get rid of it by sending it to someone?

Sounds like the holder can't even send it back to the issuer. The issuer would *have* to clawback. 

It sounds like authorization is really a direct opposite of locking, which is fine, since the semantics and legalities probably need to support both.

That leads me to wonder, though: While a lock would be expected to be temporary, I assume (again with the assumptions) that unauthorizing would be expected to be permanent. Are there any use cases where the issuer would *not* want to claw back the holder's balance after unauthorizing them?

Would it be worth automatically sending the balance back to the issuer (or adding an optional `tfMPTClawbackBalance` flag to `MPTokenAuthorize` transaction which would only be valid if `tfMPTUnauthorize` is set)?

Reply:
To clarify a bit more, unauthorization and locking work the exact same way in a sense that the holder cannot send or receive funds from the unauthorized/locked MPToken. The only difference is that when locked, the holder can still send back to the issuer. On the other hand, an "unauthorized" holder cannot even send back to the issuer. 

This decision is more of a product requirement, and they are fine with the two-step process where the issuer  unauthorizes first and then clawback.

Reply:
Makes sense. We can always add additional functionality / options later if there is community demand.